src/models/room-state.js

```diff
@@ -197,6 +197,21 @@ RoomState.prototype.setStateEvents = function(stateEvents) {
                 member.setPowerLevelEvent(event);
                 self.emit("RoomState.members", event, self, member);
             });
+
+            // Go through the sentinel members and see if any of them would be
+            // affected by the new power levels. If so, replace the sentinel.
+            for (const userId of Object.keys(self._sentinels)) {
+                const oldSentinel = self._sentinels[userId];
+                const newSentinel = new RoomMember(event.getRoomId(), userId);
+                newSentinel.setMembershipEvent(oldSentinel.events.member, self);
+                newSentinel.setPowerLevelEvent(event);
+                if (
```

Comment:
I am unsure if this condition is (a) necessary (b) a good idea.

In particular, I can see a world where a power_level event arrives to change (say) the 'notify' power_level. This doesn't change the individual members' power levels, but an application might compare the sentinel member's power level against the power_level event associated with the sentinel member, and end up with the wrong result.

In short: should we make replacing the sentinel unconditional? (Presumably at the risk of ending up with even more `RoomMember` objects floating about in the heap)

Reply:
Yeah, I was basically trying to avoid making 10,000 new sentinels whenever we put a power_levels event in HQ, but yes, it does seem a bit dicey that the sentinels will be associated with a power levels event that isn't the one actually in effect unless it specifically changed that user's power_level.

I guess we do the simpler thing for now and optimise later if needs be.